roles/common/apache2/apache2_install/templates/apache2.conf.j2

```diff
@@ -0,0 +1,45 @@
+# Include module configuration:
+Include /etc/apache2/mods-enabled/*.load
+Include /etc/apache2/mods-enabled/*.conf
+
+# Include ports listing
+Include /etc/apache2/ports.conf
+
+{% if apache2_servername is defined %}
+ServerName {{apache2_servername}}
+{% endif %}
+
+<VirtualHost {{apache2_ip}}:{{apache2_port}}>
+  ServerAdmin webmaster@localhost
+  
+  {% if apache2_documentroot is defined %}
+  DocumentRoot {{apache2_documentroot}}
+  {% endif %}
+
+  ErrorLog ${APACHE_LOG_DIR}/error.log
+  CustomLog ${APACHE_LOG_DIR}/access.log combined
+
+  {% for directory in apache2_directories %}
```

Comment:
Can you add a comment here to explain what this block is doing. It's pretty straightforward but the nested if statements mean this block needs an overview if refactoring isn't possible.